File: airesources/Java/hlt/Player.java
package hlt;

import java.util.ArrayList;
import java.lang.Exception;

public class Player {
    private int playerID;
    private int halite;
    private Shipyard shipyard;
    private ArrayList<Ship> ships;
    private ArrayList<Dropoff> dropoffs;

    public Player(int playerID_, int halite_, Shipyard shipyard_, ArrayList<Ship> ships_, ArrayList<Dropoff> dropoffs_) {
        playerID = playerID_;
        halite = halite_;
        shipyard = shipyard_;
        ships = ships_;
        dropoffs = dropoffs_;
    }

    public int getPlayerID() { return playerID; }
    public int getHalite() { return halite; }
    public Shipyard getShipyard() { return shipyard; }
    public ArrayList<Ship> getShips() { return ships; }
    public ArrayList<Dropoff> getDropoffs() { return dropoffs; }

    public void addShip(Ship ship) throws Exception {
        if(ships.contains(ship)) throw new Exception("Tried to add a duplicate ship to a player!");
        else ships.add(ship);
    }
    public void addDropoff(Dropoff dropoff) throws Exception {
        if(dropoffs.contains(dropoff)) throw new Exception("Tried to add a duplicate dropoff to a player!");
        else dropoffs.add(dropoff);
    }
}

File: starter_kits/Swift/Halite-III/Halite-III/hlt/Player.swift
//
//  Player.swift
//  Halite-III
//
//  Created by Chris Downie on 10/24/18.
//  Copyright © 2018 Chris Downie. All rights reserved.
//

import Foundation

struct Player {
    typealias ID = Int
    
    let id: ID
    let shipyard: Shipyard
    let haliteAmount: Int
    
    let ships: [Ship]
    let dropoffs: [Dropoff]
    
    // MARK: Initialization
    init(id: ID, shipyard: Shipyard, haliteAmount: Int = 0, ships: [Ship] = [], dropoffs: [Dropoff] = []) {
        self.id = id
        self.shipyard = shipyard
        self.haliteAmount = haliteAmount
        self.ships = ships
        self.dropoffs = dropoffs
    }

    // MARK: Ships
    /// Returns the ship object associated with the ship id provided as an argument.
    ///
    /// - Parameter ship: The ship ID to query for.
    /// - Returns: The ship object associated with the ship ID provided as an argument.
    func get(ship id: Ship.ID) -> Ship? {
        return ships.first(where: { ship in
            ship.id == id
        })
    }
    
    /// Returns a list of all ship objects.
    ///
    /// - Returns: All ship objects.
    func getShips() -> [Ship] {
        return ships
    }
    
    /// Checks if you have a ship with this id.
    ///
    /// - Parameter ship: The ship ID to query for.
    /// - Returns: True if you have a ship with this id.
    func has(ship id: Ship.ID) -> Bool {
        return ships.contains { $0.id == id }
    }
    
    // MARK: Dropoffs
    /// Returns the dropoff object associated with the dropoff id provided as an argument.
    ///
    /// - Parameter dropoff: The dropoff ID to query for
    /// - Returns: The dropoff object associated with the dropoff id provided as an argument.
    func get(dropoff id: Dropoff.ID) -> Dropoff? {
        return dropoffs.first(where: { $0.id == id })
    }

    /// Returns a list of all dropoff objects.
    ///
    /// - Returns: All dropoff objects.
    func getDropoffs() -> [Dropoff] {
        return dropoffs
    }
}
